include/gatekeeper_gk.h

```diff
@@ -24,13 +24,61 @@
 #include "gatekeeper_ipip.h"
 #include "gatekeeper_ggu.h"
 #include "gatekeeper_mailbox.h"
+#include "gatekeeper_lpm.h"
+
+/*
+ * The LPM supports 8-bit next hop,
+ * which at most has 128 different values.
+ */
+#define GK_MAX_NUM_POLICIES (128)
+
+#define GK_MAX_NUM_GRANTORS (4)
 
 /*
  * A flow entry can be in one of three states:
  * request, granted, or declined.
  */
 enum gk_flow_state { GK_REQUEST, GK_GRANTED, GK_DECLINED };
 
+enum gk_policy_action {
+	/* Forward the packet to a Grantor server. */
+	GK_FWD_GT,
+
+	/* Forward the packet to the back interface. */
+	GK_FWD_BCAK_NET,
+
+	/* Drop the packet. */
+	GK_DROP,
+};
+
+struct gk_policy {
+
+	enum gk_policy_action action;
+
+	/* Data that supports different policy actions. */
+	union {
+		int grantor_id;
```

Comment:
Why don't you move `struct ipip_tunnel_info`, or whatever it's needed to redirect a packet to the Grantor server, to here? You are not saving memory since field `tunnels` is statically allocated, but you'd avoid another memory access.